modules/debugging/index.qmd

```diff
@@ -95,29 +94,271 @@ detach_package(tidyverse)
+
+If you do need to use a custom dataset specific to your project, try reducing the size of the dataset before making your example. The error may only occur for some specific rows of your data. Once you've narrowed in on the issue, you can convert your minimal data object to code for producing it using the `dput()` function.
+
+For example you can use use the below code to provide the first 6 rows of the iris dataset.
+
+```{r}
+dput(head(iris))
+```
+
+Try it yourself! Create a minimal dataset using the `dput()` function for the `penguins` dataset.
+
+::: callout-tip
+## Hint
+
+We want to be putting the `penguins` data to create a minimal dataset.
+:::
+
+```{webr-teachr}
+## REVISIT THIS :( 
+# Load the penguins dataset
+library(palmerpenguins)
+# Use dput() to create a minimal dataset
+<<dput(head(penguins))>>
   
 ???
   
+"Did you use dput() with head(penguins)?" != search_ast(.code, .expr = dput(head(penguins)))
+
 ```
 
+You can then copy this generated code, and make it part of your example to create your example data.
+
+Alternatively you can save this minimal dataset to a file, and attach the file in your question.
+
+When you read that file in, make sure you avoid absolute paths to it. Not everyone has the same file structure as you, so they won't be able to access your dataset if it is stored at `/home/learnr/Documents/My Job/project_2040/data/example.csv`. Instead provide a short relative oath, such as `example.csv` or `data/example.csv`. If you're uploading files, it's also nice to provide a zip that contains a R script of your minimal reproducible code, along with the data in the folders referenced by the script.
+
+-- potentially a multiple choice question here on relative paths? ---
+
+## Provide minimal code
+
+As part of a data analysis you probably write a lot of code. You need to read in the data, tidy it up, run some calculations and make some plots. When you run into an error, providing your entire analysis just makes it more time consuming for helpers to run your code and identify the source of the error.
+
+A *minimal* reproducible example cuts out as much of the code as possible, keeping just a few essential lines for producing the error. If your error comes after the data cleaning, remove that code and start with a clean, minimal dataset.
+
+If your code includes any randomisation, it’s a good idea to set the random seed using the `set.seed()` function. This ensures that each time the code is ran, the same random results are obtained.
+
+::: callout-tip
+## What is set.seed()?
```

Comment:
```suggestion
## Reproducing random results
```

With this I think you could move the earlier `set.seed()` sentences inside.

Open to your opinion if you think this is a good change.

Reply:
I like the title change - but I think it might be nice to have randomisation as part of the actual section but the tip as a bit of a further explanation on it? To break up the text a bit as well